### PR TITLE
Basic UI: escape HTML characters

### LINKED ISSUE #3744: Basic UI: bad refresh of value containing XML
The value containing XML seems to be correctly displayed, even if truncated, when the page is first opened but when an update is sent the dispayed value rhen looks like an empty string.
Can be checked with the zonegroup channel of the Sonos binding

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #3744

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@resetnow : this one is for you ,)
All my tests are positive.

File: extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/render/SwitchRenderer.java [hunk 1]
```diff
@@ -82,7 +82,8 @@ public boolean canRender(Widget w) {
                 String button = getSnippet("button");
                 button = StringUtils.replace(button, "%item%", w.getItem());
                 button = StringUtils.replace(button, "%cmd%", mapping.getCmd());
```
Comment (resetnow) [inline extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/render/SwitchRenderer.java]: I think we need escape here as well. Test case:

```
Switch item=DemoString mappings=["value__1"="description_1", "value_\"_2"="description_2"]
```

Comment (resetnow) [inline extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/render/SwitchRenderer.java]: Also, using `StringEscapeUtils.escapeHTML` here is probably not the right choice — only quotes and backslashes need to be prepended with `\` so resulting HTML attribute has correct value when read from Javascript.

Comment (lolodomo) [inline extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/render/SwitchRenderer.java]: I thought about mappings from number or ON/OFF/OPEN/CLOSE... for example but not from a general string.
You're right, it has to be tested and fixed for general string commands too.

**Comment by lolodomo** [comment]
Fixed.

**Comment by sjka** [comment]
Thanks!

**Pull request merged**